cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/services/http/handlers/ProgramLifecycleHttpHandlerTest.java

```diff
@@ -440,10 +446,13 @@ public void testProgramStartStopStatusErrors() throws Exception {
   }
 
   /**
-   * Tests history of a mapreduce.
+   * Tests history of a mapreduce
+   * TODO: These tests will be modified after LCM changes to the schedules and it's apis.
```

Comment:
does this test do anything with schedules? If not it shouldn't be ignored.

Reply:
Yes, they do. I'll post the stack trace here in a bit. These tests are ignored only for now - they will be modified once we make api changes in ProgramLifecycleHandler

Reply:
So these functions call `startProgram` in `ProgramLifecycleHttpHandler (
https://github.com/cdapio/cdap/blob/develop/cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/services/http/AppFabricTestBase.java#L882)

```
java.lang.AssertionError: Wrong response code with message 'program:testnamespace2.dummy.1.0.0.mapreduce.dummy-batch' was not found. 
Expected :200
Actual   :404
<Click to see difference>


	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:743)
	at org.junit.Assert.assertEquals(Assert.java:118)
	at org.junit.Assert.assertEquals(Assert.java:555)
	at io.cdap.cdap.internal.app.services.http.AppFabricTestBase.assertResponseCode(AppFabricTestBase.java:1490)
	at io.cdap.cdap.internal.app.services.http.AppFabricTestBase.startProgram(AppFabricTestBase.java:883)
	at io.cdap.cdap.internal.app.services.http.AppFabricTestBase.startProgram(AppFabricTestBase.java:867)
	at io.cdap.cdap.internal.app.services.http.AppFabricTestBase.startProgram(AppFabricTestBase.java:874)
	at io.cdap.cdap.internal.app.services.http.handlers.ProgramLifecycleHttpHandlerTest.verifyProgramHistory(ProgramLifecycleHttpHandlerTest.java:1736)
	at io.cdap.cdap.internal.app.services.http.handlers.ProgramLifecycleHttpHandlerTest.testHistory(ProgramLifecycleHttpHandlerTest.java:1730)
	at io.cdap.cdap.internal.app.services.http.handlers.ProgramLifecycleHttpHandlerTest.testMapreduceHistory(ProgramLifecycleHttpHandlerTest.java:457)```